### PR TITLE
toast replaced with snackbar

### PR DESCRIPTION
**Pull request by criticalAY**
## Purpose / Description
toast replacement

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by criticalAY** [comment]
` @KotlinCleanup("check RTL with concat")` is this needed there now because the toast is gone ?

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt [hunk 1]
```diff
@@ -416,17 +416,11 @@ open class AnkiActivity : AppCompatActivity, SimpleMessageDialogListener, Collec
         }
     }
 
-    @KotlinCleanup("toast -> snackbar")
     open fun openUrl(url: Uri) {
         // DEFECT: We might want a custom view for the toast, given i8n may make the text too long for some OSes to
         // display the toast
         if (!AdaptionUtil.hasWebBrowser(this)) {
-            @KotlinCleanup("check RTL with concat")
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:424]: keep the cleanup. It is about if the string is concatened correctly on RTL languages.

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt [hunk 2]
```diff
@@ -416,17 +416,11 @@ open class AnkiActivity : AppCompatActivity, SimpleMessageDialogListener, Collec
...
-            showThemedToast(
-                this,
-                resources.getString(R.string.no_browser_notification) + url,
-                false
-            )
+            showSnackbar(resources.getString(R.string.no_browser_notification) + url, Snackbar.LENGTH_LONG)
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:423]: I'd extract `resources.getString(R.string.no_browser_notification) + url` into a variable to make this more readable. `noBrowserMessage` seems good to me

**Review by BrayanDSO — CHANGES_REQUESTED** [review]
Tests fail

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt (hunk 1 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:424]: cool

**Comment by criticalAY** [comment]
`java.lang.IllegalArgumentException: While trying to show a snackbar, could not find a view with id root_layout in com.ichi2.anki.dialogs.utils.FragmentTestActivity@292bfa01` the test error is caused by this but what to pass there ? It is working fine

**Comment by criticalAY** [comment]
@BrayanDSO I am reverting the code and removing the annotation for the cleanup as there is no root layout to attach the snackbar. So its better to keep the toast instead

**Comment by BrayanDSO** [comment]
Have you tested this piece of code on a real device? 

I believe that the only issue with this is the unit tests, which should be fixed instead of just removing the TODO. Perhaps by using a CoordinatorLayout somewhere

**Comment by BrayanDSO** [comment]
> I have tested it and it worked fine but the test failed because of the layout issue

It may be tough, but the path here is to fix the test

**Comment by criticalAY** [comment]
For that we need to locate the test file could you help me with that I would like to give it a try

**Comment by criticalAY** [comment]
I forced push back then and ended up here will try to get it fixed because this PR is of less relevance because there needs to be some edits in test case for this snackbar to pass the tests. But surely try fixing it and I will keep rebasing in mind in the future for sure

**Comment by criticalAY** [comment]
I have corrected this PR but would like to know how to fix the tests @BrayanDSO because it is working fine at my end doesn't seems to be an issue

**Comment by criticalAY** [comment]
Screenshot_13

**Comment by mikehardy** [comment]
@criticalAY please learn how to rebase and rebase. The commit history here is not good

**Comment by criticalAY** [comment]
> @criticalAY please learn how to rebase and rebase. The commit history here is not good

done Mike

**Comment by criticalAY** [comment]
Mike is there a way to drop multiple commits in one go I had to place a d before I guess 50 commits to clean this branch

**Comment by mikehardy** [comment]
Once you've done a "merge" it's difficult. I suggest pulling the actual commits that do things you are interested in doing, in this PR, off to the side (git reset --soft HEAD~'some number' can probably do that followed by git stash)

Once you are *sure* you have the changes you actually want, that you intend for this PR, stashed, then git reset --hard upstream/main to just purge your local branch state. Then git stash pop, git commit your changes as a single nice commit, and push --force-with-lease

But, there are lots of ways to do it. End goal is to just have the actual textual changes that implement whatever you implementing as a nice single commit, as opposed to a trail of like 20 with 3 of them being merges - those are impossible to follow along with, as a reviewer

**Comment by criticalAY** [comment]
```
    fun testNormalStartupSelectingItem() {
        val linkedItem = getItemLinkingTo(R.string.link_manual)
        val v = getRecyclerViewFor(linkedItem)
        clickChildAtIndex(v, 0)
        MatcherAssert.assertThat(activity!!.lastUrlOpened, Matchers.equalTo(getResourceString(R.string.link_manual)))
    }
```

There is this test function I would like to know if we really need the `!!` here can be prefer `?`

**Comment by BrayanDSO** [comment]
I'd try setting a layout with a `CoordinatorLayout` on `FragmentTestActivity.kt`

**Comment by criticalAY** [comment]
> I'd try setting a layout with a `CoordinatorLayout` on `FragmentTestActivity.kt`

```
val coordinatorLayout = findViewById<CoordinatorLayout>(R.id.coordinator_layout)
 assertNotNull(coordinatorLayout)
```

something like this?

**Comment by criticalAY** [comment]
@BrayanDSO The tests passed when added the layout I have added the DeckPicker layout right now but can we consider creating a new layout because I am not sure if the current commit is a best practice

**Pull request closed without merging**